harness/determined/cli/_color_json.py

```diff
@@ -0,0 +1,92 @@
+"""
+colorize JSON output.
+test string:
+   echo '{"a": "b", "c": ["1", 2.0, 3.1, 5, "4", {"d": true, "e": null, "f": [1, 2, 3]}]}'
+"""
+
+import json
+from typing import Any, TextIO
+
+from termcolor import colored
+
+KEY = "blue"
+PRIMITIVES = "white"
+SEPARATORS = "yellow"
+STRING = "green"
+
+
+def render_json(obj: Any, out: TextIO, indent: str = "  ", sort_keys: bool = False) -> None:
```

Comment:
Slight preference for 

```python
def render_json(obj, indent: str = "  ", sort_keys: bool = False) -> str:
```

because pure functions are a little easier to think about and test. Also brings the function's behavior a little more in line with its name, and might make the later conditional a little easier.

What prompted you to pass a `TextIO`?

Reply:
I believe it was my original code that did this.

I also have only a slight preference.  I did it this way because I think converting stream operators into string operators is an antipattern in general.  Probably not a huge deal since the input json already has to fit in memory.  But this is more similar to e.g. `json.dump(f)`, which I would prefer instead of `f.write(json.dumps())`, for example.

Unit-testability of either option is pretty trivial I think.